hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -401,57 +402,29 @@ public static HoodieTableConfig loadFromHoodieProps(HoodieStorage storage, Stora
   }
 
   private HoodieTableConfig(HoodieStorage storage, StoragePath metaPath) {
-    this(storage, metaPath, null, null, null, false);
+    this(storage, metaPath, null, null, null);
   }
 
   public HoodieTableConfig(HoodieStorage storage, StoragePath metaPath, RecordMergeMode recordMergeMode, String payloadClassName,
                            String recordMergeStrategyId) {
-    this(storage, metaPath, recordMergeMode, payloadClassName, recordMergeStrategyId, true);
```

Comment:
From usability standpoint, could we add a config to allow payload class or merge mode/strategy update when enabled, and keep this config disabled by default, i.e., not allowing record merge mode/strategy to change unless user enables it in rare cases?